src/workerd/api/data-url-test.c++

```diff
@@ -0,0 +1,632 @@
+      kj::heapArray<kj::byte>({87, 65}),
+    },
+    {
+      "data:x;base64;x,WA"_kj,
+      "text/plain;charset=US-ASCII"_kj,
+      kj::heapArray<kj::byte>({87, 65}),
+    },
+    {
+      "data:x;base64=x,WA"_kj,
+      "text/plain;charset=US-ASCII"_kj,
+      kj::heapArray<kj::byte>({87, 65}),
+    },
+    {
+      "data:; base64,WA"_kj,
+      "text/plain;charset=US-ASCII"_kj,
+      kj::heapArray<kj::byte>({88}),
+    },
+    {
+      "data:;  base64,WA"_kj,
+      "text/plain;charset=US-ASCII"_kj,
+      kj::heapArray<kj::byte>({88}),
+    },
+    {
+      "data:  ;charset=x   ;  base64,WA"_kj,
+      "text/plain;charset=x"_kj,
+      kj::heapArray<kj::byte>({88}),
+    },
+    {
+      "data:;base64;,WA"_kj,
+      "text/plain"_kj,
+      kj::heapArray<kj::byte>({87, 65}),
+    },
+    {
+      "data:;base64 ,WA"_kj,
+      "text/plain;charset=US-ASCII"_kj,
+      kj::heapArray<kj::byte>({88}),
+    },
+    {
+      "data:;base64   ,WA"_kj,
+      "text/plain;charset=US-ASCII"_kj,
+      kj::heapArray<kj::byte>({88}),
+    },
+    {
+      "data:;base 64,WA"_kj,
+      "text/plain"_kj,
+      kj::heapArray<kj::byte>({87, 65}),
+    },
+    {
+      "data:;BASe64,WA"_kj,
+      "text/plain;charset=US-ASCII"_kj,
+      kj::heapArray<kj::byte>({88}),
+    },
+    {
+      "data:;%62ase64,WA"_kj,
+      "text/plain"_kj,
+      kj::heapArray<kj::byte>({87, 65}),
+    },
+    {
+      "data:%3Bbase64,WA"_kj,
+      "text/plain;charset=US-ASCII"_kj,
+      kj::heapArray<kj::byte>({87, 65}),
+    },
+    {
+      "data:;charset=x,X"_kj,
+      "text/plain;charset=x"_kj,
+      kj::heapArray<kj::byte>({88}),
+    },
+    {
+      "data:; charset=x,X"_kj,
+      "text/plain;charset=x"_kj,
+      kj::heapArray<kj::byte>({88}),
+    },
+    {
+      "data:;charset =x,X"_kj,
+      "text/plain"_kj,
+      kj::heapArray<kj::byte>({88}),
+    },
+    {
+      "data:;charset= x,X"_kj,
+      "text/plain;charset=\" x\""_kj,
+      kj::heapArray<kj::byte>({88}),
+    },
+    {
+      "data:;charset=,X"_kj,
+      "text/plain"_kj,
+      kj::heapArray<kj::byte>({88}),
+    },
+    {
+      "data:;charset,X"_kj,
+      "text/plain"_kj,
+      kj::heapArray<kj::byte>({88}),
+    },
+    {
+      "data:;charset=\"x\",X"_kj,
+      "text/plain;charset=x"_kj,
+      kj::heapArray<kj::byte>({88}),
+    },
+    {
+      "data:;CHARSET=\"X\",X"_kj,
+      "text/plain;charset=X"_kj,
+      kj::heapArray<kj::byte>({88}),
+    }
+  };
+
+  auto testPtr = kj::arrayPtr<Test>(tests, 72);
+
+  for (auto& test : testPtr) {
+    if (test.mimeType == nullptr) {
+      KJ_ASSERT(DataUrl::tryParse(test.input) == kj::none);
+    } else {
+      auto parsed = KJ_ASSERT_NONNULL(DataUrl::tryParse(test.input));
+      KJ_ASSERT(parsed.getMimeType().toString() == test.mimeType);
+      KJ_ASSERT(parsed.getData() == test.data);
+    }
+  }
+}
+
+struct Base64Test {
+  kj::StringPtr input;
+  kj::Array<kj::byte> expected;
+};
+
+KJ_TEST("DataUrl base64") {
```

Comment:
Did we get these test cases from somewhere or did we write them ourselves? Can we add a comment or test name that specifies this?